worker/buildbot_worker/commands/transfer.py

```diff
@@ -107,7 +107,8 @@ def start(self):
         self._reactor.callLater(0, self._loop, d)
 
         def _close_ok(res):
-            self.fp = None
+            if self.fp:
+                self.fp.close()
```

Comment:
Maybe you should set self.fp = None
after closing it?  Maybe not 100% necessary, but way not?